src/components/todoApp.js

```diff
@@ -21,19 +19,21 @@ export default class TodoApp extends React.Component {
 					<TodoEntry todoModel={todoModel} />
 				</header>
 				<TodoOverview todoModel={todoModel} viewModel={viewModel} />
-				<TodoFooter todoModel={todoModel} viewModel={viewModel} />
+				<TodoFooter
+					todoModel={todoModel}
+					viewModel={viewModel}
+					stateNavigator={stateNavigator}
+				/>
 			</div>
 		);
 	}
 
 	componentDidMount() {
 		var viewModel = this.props.viewModel;
-		var router = Router({
-			'/': function() { viewModel.todoFilter = ViewModel.ALL_TODOS; },
-			'/active': function() { viewModel.todoFilter = ViewModel.ACTIVE_TODOS; },
-			'/completed': function() { viewModel.todoFilter = ViewModel.COMPLETED_TODOS; }
-		});
-		router.init('/');
+		var stateNavigator = this.props.stateNavigator;
+		stateNavigator.states.todo.navigated = function(data) {
+			viewModel.todoFilter = data.filter;
+		}
```

Comment:
The `navigated` function attached to the `todo` is called every time the Url changes. The name `todo` is taken from the key assigned in the router configuration. The `data` passed in holds the new filter value.